### PR TITLE
[FIX] mrp: fix singleton error while multiple moves in 'move_finished_ids'

### PR DESCRIPTION
**Pull request by omra-odoo**
ValueError: Expected singleton: stock.move(32, 33)
This error occurs when we get the multiple lines of the same products in 'finish_moves' and after that when we try to mark as done at that time this error occurs.

see stack trace: 
```
ValueError: too many values to unpack (expected 1)
  File "odoo/models.py", line 5199, in ensure_one
    _id, = self._ids
ValueError: Expected singleton: stock.move(32, 33)
  File "odoo/http.py", line 2116, in __call__
    response = request._serve_db()
  File "odoo/http.py", line 1698, in _serve_db
    return service_model.retrying(self._serve_ir_http, self.env)
  File "odoo/service/model.py", line 134, in retrying
    result = func()
  File "odoo/http.py", line 1725, in _serve_ir_http
    response = self.dispatcher.dispatch(rule.endpoint, args)
  File "odoo/http.py", line 1923, in dispatch
    result = self.request.registry['ir.http']._dispatch(endpoint)
  File "addons/website/models/ir_http.py", line 234, in _dispatch
    response = super()._dispatch(endpoint)
  File "odoo/addons/base/models/ir_http.py", line 154, in _dispatch
    result = endpoint(**request.params)
  File "odoo/http.py", line 715, in route_wrapper
    result = endpoint(self, *args, **params_ok)
  File "addons/web/controllers/dataset.py", line 32, in call_button
    action = self._call_kw(model, method, args, kwargs)
  File "addons/web/controllers/dataset.py", line 24, in _call_kw
    return call_kw(request.env[model], method, args, kwargs)
  File "odoo/api.py", line 461, in call_kw
    result = _call_kw_multi(method, model, args, kwargs)
  File "odoo/api.py", line 448, in _call_kw_multi
    result = method(recs, *args, **kwargs)
  File "home/odoo/src/enterprise/saas-16.2/quality_mrp/models/mrp_production.py", line 51, in button_mark_done
    return super(MrpProduction, self).button_mark_done()
  File "addons/mrp/models/mrp_production.py", line 1791, in button_mark_done
    productions_not_to_backorder._post_inventory(cancel_backorder=True)
  File "addons/mrp/models/mrp_production.py", line 1508, in _post_inventory
    if finish_moves and not finish_moves.quantity_done:
  File "odoo/fields.py", line 1151, in __get__
    record.ensure_one()
  File "odoo/models.py", line 5202, in ensure_one
    raise ValueError("Expected singleton: %s" % self)
```

applying these changes will resolve this issue.

sentry-4217897073

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

**Comment by omra-odoo** [comment]
Hello @odoo/sentry_reviewers 

can you please check this PR

Thanks!

File: addons/mrp/models/mrp_production.py [hunk 1]
```diff
@@ -1523,9 +1523,9 @@ def _post_inventory(self, cancel_backorder=False):
         for order in self:
             finish_moves = order.move_finished_ids.filtered(lambda m: m.product_id == order.product_id and m.state not in ('done', 'cancel'))
             # the finish move can already be completed by the workorder.
-            if finish_moves and not finish_moves.quantity_done:
-                finish_moves._set_quantity_done(float_round(order.qty_producing - order.qty_produced, precision_rounding=order.product_uom_id.rounding, rounding_method='HALF-UP'))
-                finish_moves.move_line_ids.lot_id = order.lot_producing_id
+            for finish_move in finish_moves.filtered(lambda m: not m.quantity_done):
+                finish_move._set_quantity_done(float_round(order.qty_producing - order.qty_produced, precision_rounding=order.product_uom_id.rounding, rounding_method='HALF-UP'))
+                finish_move.move_line_ids.lot_id = order.lot_producing_id
```
Comment (ticodoo) [inline addons/mrp/models/mrp_production.py:1528]:
```suggestion
            for move in finish_moves:
                if move.quantity_done:
                    continue
                move._set_quantity_done(float_round(order.qty_producing - order.qty_produced, precision_rounding=order.product_uom_id.rounding, rounding_method='HALF-UP'))
                move.move_line_ids.lot_id = order.lot_producing_id
```

**Review by ticodoo — COMMENTED** [review]
Is it even possible for this error to occur in standard Odoo? It shouldn't be possible for:

`finish_moves = order.move_finished_ids.filtered(lambda m: m.product_id == order.product_id and m.state not in ('done', 'cancel'))`

to result in len(finish_moves) > 1. If it is, then the calculations that are being applied to the multiple moves probably won't correct with the code as is (since it's designed for 1 move)

File: addons/mrp/models/mrp_production.py (hunk 1 above)
Comment (ticodoo) [reply addons/mrp/models/mrp_production.py:1528]: @omra-odoo ok, apply my suggested change and I will merge

Comment (omra-odoo) [reply addons/mrp/models/mrp_production.py]: @ticodoo changes have been done.

Thanks!

**Comment by ticodoo** [comment]
@robodoo r+

note that in theory this use case should never happen so this fix will most likely lead to an incorrect value in the strange chance that customizations or something weird causes it

**Pull request closed without merging**